[박민서] 5주차 과제 제출

m2nsp opened the pull request:
<!--
제목 예시: [이름] n주차 과제 제출
-->

## 📌 구현 기능  
- [x] 게시판 CRUD
- [x] 게시글 CRUD  

## 🗂️ 과제 정리  
- 게시판 생성
![스크린샷 2025-04-12 000501](https://github.com/user-attachments/assets/0b5cff13-d5bb-4e30-b54e-5b1552b6ed24)

- 게시판 수정
![스크린샷 2025-04-12 001054](https://github.com/user-attachments/assets/2e6331ae-1d2a-45fc-ae01-05facb616ff4)

- 게시판 조회
![스크린샷 2025-04-12 000606](https://github.com/user-attachments/assets/e79c88f7-9533-462f-9dac-9b5e14483dc2)

![스크린샷 2025-04-12 000628](https://github.com/user-attachments/assets/16af9bba-746e-428e-8bcd-4c368ee93450)

- 게시판 삭제
![스크린샷 2025-04-12 001115](https://github.com/user-attachments/assets/c311791b-0da4-4a69-8fb6-7f1cf585646f)

- 게시글 생성
![스크린샷 2025-04-12 015226](https://github.com/user-attachments/assets/3b23d7cb-de28-4f91-a289-9c6e1f23f0f1)

- 게시글 수정
![스크린샷 2025-04-12 021354](https://github.com/user-attachments/assets/19c1b630-b756-4b26-814d-187ca9b4db2f)

- 게시글 목록 조회
![스크린샷 2025-04-12 015359](https://github.com/user-attachments/assets/145599ab-5995-4f81-8475-1271c7cb8bd7)


- 게시글 상세 조회
![스크린샷 2025-04-12 015800](https://github.com/user-attachments/assets/3e991617-f5f2-4118-813a-15f74cc6d704)

- 게시글 삭제
![스크린샷 2025-04-12 020746](https://github.com/user-attachments/assets/dd0dd3ce-4172-4e3e-ad6c-70fe063193f7)

## ⚠️ 어려웠던 점  
- 지금 게시글 생성할때는 제대로 들어가는데 수정하면 null이 되어버리고.. createdAt, updatedAt도 어노테이션써서 세션에서 한 대로 한것 같은데 null로 들어가는데 왜 그런지 모르겠습니다...ㅠ

m2nsp committed:
✨feat: 게시판 CRUD

m2nsp committed:
Update README.md

m2nsp committed:
✨feat: 게시글 CRUD

topograp2 commented on src/main/java/efub/assignment/community/board/controller/BoardController.java:
```diff
@@ -0,0 +1,59 @@
+
+    private final BoardService boardService;
+
+    public BoardController(BoardService boardService) {
```
어노테이션을 사용하여  RequiredArgsConstructor로 처리해도 괜찮을 것 같아용!😁

topograp2 commented on src/main/java/efub/assignment/community/board/controller/BoardController.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    // 게시판 생성
+    @PostMapping("/")
```
@PostMapping만 쓰는 게 더 깔끔할 것 같아용😊😊

topograp2 commented on src/main/java/efub/assignment/community/board/domain/Board.java:
```diff
@@ -0,0 +1,91 @@
+        return new Board(boardName, announcement, description, master);
+    }
+
+    //Getter
```
Getter는 @Getter 어노테이션 쓰면 깔끔할 것 같아요!!

topograp2 commented on src/main/java/efub/assignment/community/board/domain/Board.java:
```diff
@@ -0,0 +1,91 @@
+    }
+
+    // private 생성자
+    public Board(String boardName, String announcement, String description, Member master) {
```
Builder 패턴 적용하면 더 좋을 것 같아요!!👍👍

topograp2 commented on src/main/java/efub/assignment/community/board/dto/Request/BoardRequestDTO.java:
```diff
@@ -0,0 +1,37 @@
+        this.masterId = masterId;
+    }
+
+    //Getter
```
마찬가지로 Getter 어노테이션 이용하면 깔끔할 것 같아요 :)

topograp2 commented on src/main/java/efub/assignment/community/board/dto/Request/BoardMasterRequestDTO.java:
```diff
@@ -0,0 +1,13 @@
+        return masterId;
+    }
+
+    public void setMasterId(Long masterId){
```
없어도 되지 않을까 싶습니당

topograp2 commented on src/main/java/efub/assignment/community/board/dto/Request/BoardRequestDTO.java:
```diff
@@ -0,0 +1,37 @@
+
+    public BoardRequestDTO() {};
+
+    public BoardRequestDTO(String boardName, String announcement, String description, Long masterId){
```
해당 생성자는 @AllArgsConstructor 쓰면 될 것 같습니당

topograp2 commented on src/main/java/efub/assignment/community/board/domain/Board.java:
```diff
@@ -0,0 +1,91 @@
+
+
+    // 기본 생성자 (JPA가 필요로 함)
+    protected Board() {
```
@NoArgsConstructor(access = AccessLevel.PROTECTED) 이렇게 바꾸셔도 좋을 것 같아요!

topograp2 commented on src/main/java/efub/assignment/community/board/dto/Response/BoardInfoResponseDTO.java:
```diff
@@ -0,0 +1,51 @@
+import efub.assignment.community.board.domain.Board;
+
+// 단일 게시판 조회 시 정보 반환 DTO
+public class BoardInfoResponseDTO {
```
Getter, Builder, NoArgsConstructor, AllArgsConstructor 이용하시면 코드 수를 훨씬 줄일 수 있을 것 같습니닷!
from 메서드도 builder로 변경해서 적어도 좋을 거 같아용 😊

topograp2 commented on src/main/java/efub/assignment/community/post/service/PostService.java:
```diff
@@ -0,0 +1,106 @@
+    //게시글 생성
+    @Transactional
+    public PostResponseDTO createPost(Long boardId, PostRequestDTO requestDTO){
+        //1. 게시판 유효성 검사
```
이 부분은 아래 코드에서도 중복적으로 사용되고 있는 메서드 같아서 따로 findByBoardId로 메서드를 만들어서 사용하는 것도 좋을 것 같아요!

topograp2 commented on src/main/java/efub/assignment/community/board/service/BoardService.java:
```diff
@@ -0,0 +1,80 @@
+    // 게시판 생성
+    @Transactional
+    public BoardResponseDTO createBoard(BoardRequestDTO requestDTO){
+        Member master = memberRepository.findById(requestDTO.getMasterId())
```
중복적으로 사용되는 코드는 메서드로 따로 빼서 사용하는 게 좋을 것 같아요!!

topograp2 commented on src/main/java/efub/assignment/community/post/domain/Post.java:
```diff
@@ -0,0 +1,77 @@
+    @Column
+    private String content;
+
+    @CreatedDate @Column(updatable = false)
```
`@CreatedDate`, `@LastModifiedDate`는 JPA Auditing 기능이여서 해당 어노테이션을 사용하기 위해서는 엔티티 클래스에 @EntityListeners(AuditingEntityListener.class)를 붙여주어야 합니다 😊또한, `@SpringBootApplication`이 존재하는 클래스에 `@EnableJpaAuditing`을 추가해야 제대로 작동합니당

근데 이렇게 CreatedData, LastModifiedDate를 사용할 때마다 엔티티리스너를 붙이는 과정 등이 중복되고 귀찮으니까 실습시간에 했던 것처럼 BaseEntity로 따로 빼서 해당 클래스에 BaseEntity를 상속받게 하는 게 좀 더 편할 것 같아요!!

topograp2 reviewed (approved):
안녕하세요! 민서님~ 코드리뷰 남겨드렸습니다! 저도 아직 배우는 입장이라서 미숙한 부분이 있을 수 있지만 참고 삼아서 읽어주시면 감사할 것 같아요!

 다른 건 다 좋은데 전반적으로 생성자 어노테이션, 빌더패턴 적용, Getter 어노테이션 등을 추가해주시면 더 깔끔한 코드가 되지 않을까 싶습니다!

그리고 _왜 수정 시에 null이 되는지 모르겠다_고 하셨는데 제 생각에는 **postController에서 requsetDTO에 requestBody 어노테이션이 없어서** 그런 것 같습니다!!ㅜㅜ 그래서 지금 받아올 때 해당 requestDTO에 대해 postman으로 요청을 해도 다 null로 보내주고 있습니다! 또 수정하는 것도 PostRequestDTO를 재사용하기 보다는 PostUpdateRequestDTO로 따로 DTO를 만드시는 게 좋을 것 같아요

시험기간인데 코드 작성해주시느라 수고 많으셨고 시험 파이팅입니다 👍👍

topograp2 commented on src/main/java/efub/assignment/community/post/controller/PostController.java:
```diff
@@ -0,0 +1,55 @@
+        return ResponseEntity.status(HttpStatus.CREATED).body(responseDTO);
+    }
+
+    // 게시글 목록 조회
```
여기서 지금 익명으로 게시글을 작성해도 writerId가 들어가게 되는데 익명 작성 시에는 writerId를 보내주지 않는 게 더 맞는 것 같아요!!

topograp2 commented on src/main/java/efub/assignment/community/post/service/PostService.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+    //게시글 목록 조회
+    @Transactional(readOnly = true)
```
여기서 지금 익명으로 작성된 게시글도 회원아이디를 보내주게 되어있는데 익명 작성시에는 writerId를 보내주지 않게 하는 게 좋을 것 같습니다!!

m2nsp committed:
♻️refactor: week5 - 코드리뷰 기반 리팩토링

- createdAt, updatedAt baseEntity 분리
- 게시물 수정 @ResponseBody 어노테이션 추가
- 중복된 로직 함수로 변환
